### PR TITLE
feat: Add sentry widget that includes other sentry widgets

### LINKED ISSUE #1839: Add a central `SentryWidget`
### Description

Let's explore the possibility to add a central `SentryWidget` that allows to enable/disable features of other widgets such as `SentryScreenshotWidget` etc... so we avoid causing a nested Widget structure for the user.

This also helps in eliminating the need for adding new widgets for cases such as https://github.com/getsentry/sentry-dart/issues/1832 since we could potentially just add it to the `SentryWidget`

### PR DESCRIPTION
**Pull request by buenaflor**
## :scroll: Description

`SentryWidget` serves as a container widget that reduces the amount of widget nesting for the user and makes adding functionalities that depend on widgets much easier

## :bulb: Motivation and Context

Closes #1839 

## :green_heart: How did you test it?
Manual
Unit Test

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by kuhnroyal** [comment]
I would suggest to even go further and make the existing widgets check their appropriate options themselves and then always add them to the widget tree. If the options are disabled, they can just `return child`. The widgets don't hurt but the code is clearer and more maintainable and could be tested individually.

Additionally, returning the widgets from functions could lead to unnecessary rebuilds.
Good summary here: https://www.youtube.com/watch?v=IOyq-eTRhvo

**Comment by buenaflor** [comment]
Good suggestions, thx for the hint

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 ### Features
 
+- Add `SentryWidget` ([#1846](https://github.com/getsentry/sentry-dart/pull/1846))
+- APM for isar ([#1726](https://github.com/getsentry/sentry-dart/pull/1726))
```
Comment (krystofwoldrich) [inline CHANGELOG.md:9]: `- APM for isar ([#1726](https://github.com/getsentry/sentry-dart/pull/1726))` was released in 7.15.0. I'm not sure why it wasn't in the changelog

**Review by krystofwoldrich — APPROVED** [review]
Looks good! 🚀

I would also add a small code snippet to the changelog, but I leave that up to you.

- [x] Also we should update the docs

File: CHANGELOG.md (hunk 1 above)
Comment (buenaflor) [reply CHANGELOG.md:9]: There was a mistake. We didn't add isar to the craft file so it was never published, I removed it from the 7.15.0 changelog

**Pull request merged**